pkg/cnidel/cniconfs.go

```diff
@@ -45,25 +47,17 @@ func readCniConfigFile(netInfo *danmtypes.DanmNet) ([]byte, error) {
 func getSriovCniConfig(netInfo *danmtypes.DanmNet, ipamOptions danmtypes.IpamConfig, ep *danmtypes.DanmEp) ([]byte, error) {
   vlanid := netInfo.Spec.Options.Vlan
   sriovConfig := sriovNet {
-    Name:   netInfo.Spec.NetworkID,
-    Type:   "sriov",
-    PfName: netInfo.Spec.Options.Device,
-    IfName: ep.Spec.Iface.Name,
-    L2Mode: true,
-    Vlan:   vlanid,
-    Dpdk:   DpdkOption{},
-    Ipam:   ipamOptions,
+    Name:      netInfo.Spec.NetworkID,
+    Type:      "sriov",
+    PfName:    netInfo.Spec.Options.Device,
+    L2Mode:    true,
+    Vlan:      vlanid,
+    Ipam:      ipamOptions,
+    DeviceID:  ep.Spec.Iface.DeviceID,
```

Comment:
also, will this work without filling-in DeviceInfo struct too?
I mean I guess technically it will work because of this code: https://github.com/intel/sriov-cni/blob/master/sriov/main.go#L38, but it won't pickup the exact Device we have wanted, it will allocate one for itself from the same PF, but that's not what we want

Reply:
or maybe this is the intended behaviour from our side? :)
I will sleep on it during the weekend

Reply:
According to our tests, it works as expected. The CNI picks up the interface specified in DeviceID.
check this please: https://github.com/intel/sriov-cni/blob/c97db95e4355fa247116608dcdb0da2f18f02140/pkg/config/config.go#L45

Reply:
yes you are right!
perfect